core/src/main/java/io/undertow/server/protocol/framed/AbstractFramedStreamSinkChannel.java

```diff
@@ -240,12 +240,12 @@ public void run() {
 
     @Override
     public void shutdownWrites() throws IOException {
+        // Queue prior to shutting down writes, since we might send the write buffer
```

Comment:
Do you think we should add an assert here to validate `lock` is not already held?

Reply:
It is not necessary @carterkozak - lock is private variable in AbstractFramedStreamSinkChannel and so external classes using this method cannot obtain the lock.

Reply:
I was thinking it could prevent regressions if this class is updated in the future, but I defer to your judgement.

Reply:
In such context (defensive programming) it would make sense of course.